### PR TITLE
Docs/navigation styling

### PR DESCRIPTION
**Pull request by silasjmatson**
This PR adds documentation for the following:

* Styling
* Navigator generator
* Navigation
* Navigation helpers
* Testing [stub for now]

We can merge it if it looks good.

### CODE REVIEW

**Review by jamonholmgren — APPROVED** [review]
Let's get this merged in as-is, and move forward with more documentation efforts.

**Comment by jamonholmgren** [comment]
Actually, I'll just work on this branch as-is.

File: boilerplate/app/navigators/app-navigator.tsx [hunk 1]
```diff
@@ -53,7 +53,11 @@ interface NavigationProps extends Partial<React.ComponentProps<typeof Navigation
 
 export const AppNavigator = (props: NavigationProps) => {
   const colorScheme = useColorScheme()
-  useBackButtonHandler(canExit)
+
+  // What route names do we allow the back button to exit the app from?
+  const exitRoutes = ["welcome"]
+  useBackButtonHandler((routeName) => exitRoutes.includes(routeName))
```
Comment (jamonholmgren) [inline boilerplate/app/navigators/app-navigator.tsx:59]: Simplifying this file a bit, since the `canExit` function is no longer used anywhere else.

**Review by mazenchami — APPROVED** [review]
LGTM! 🚀

**Pull request merged**